test/sample.jl

```
@@ -184,14 +185,36 @@
             MySampler(),
             MCMCThreads(),
             3,
-            100;
+            nchains;
             progress=false,
-            init_params=Iterators.repeated(init_params),
+            init_params=Iterators.repeated(init_params, nchains),
         )
-        @test length(chains) == 100
+        @test length(chains) == nchains
         @test all(
             chain[1].a == init_params.a && chain[1].b == init_params.b for chain in chains
         )
+
+        # Too many `init_params`
+        @test_throws ArgumentError sample(
+            MyModel(),
+            MySampler(),
+            MCMCThreads(),
+            3,
+            nchains;
+            progress=false,
+            init_params=Iterators.repeated(init_params, nchains + 1),
```

So `_first_or_nothing` doesn't actually care if there are too many initial parameters :confused: 

IMO it seems like we want to raise an error in this case since it's otherwise very easy to accidentally do the incorrect thing.